Remove the gatekeeping to set a non-executable stack for libunwind,
as we don't define any of these directives but still want it.

diff -Nru a/src/assembly.h b/src/assembly.h
--- a/src/assembly.h    2016-10-23 21:48:47.000000000 +0000
+++ b/src/assembly.h    2017-05-16 12:38:47.209169931 +0000
@@ -47,12 +47,7 @@
 #define SYMBOL_IS_FUNC(name) .type name,@function
 #endif
 
-#if defined(__GNU__) || defined(__ANDROID__) || defined(__FreeBSD__) || \
-    defined(__Fuchsia__)
 #define NO_EXEC_STACK_DIRECTIVE .section .note.GNU-stack,"",%progbits
-#else
-#define NO_EXEC_STACK_DIRECTIVE
-#endif
 
 #else
 
